staging/src/k8s.io/apiserver/pkg/util/webhook/client.go

```diff
@@ -14,32 +14,40 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package config
+package webhook
 
 import (
 	"context"
-	"encoding/json"
 	"errors"
 	"fmt"
 	"net"
 	"net/url"
 
-	lru "github.com/hashicorp/golang-lru"
-	admissionv1beta1 "k8s.io/api/admission/v1beta1"
-	"k8s.io/api/admissionregistration/v1beta1"
+	"github.com/hashicorp/golang-lru"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/runtime/serializer"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
-	webhookerrors "k8s.io/apiserver/pkg/admission/plugin/webhook/errors"
 	"k8s.io/client-go/rest"
 )
 
 const (
 	defaultCacheSize = 200
 )
 
+// ClientConfigInterface is the interface to access webhook client config to share webhook connect code between
+// admission webhooks and CRD conversion webhooks.
+type ClientConfigInterface interface {
+	GetURL() *string
+	GetCABundle() []byte
+	GetServiceName() *string
+	GetServiceNamespace() *string
+	GetServicePath() *string
+	GetCacheKey() (string, error)
+}
+
 var (
-	ErrNeedServiceOrURL = errors.New("webhook configuration must have either service or URL")
+	errNeedServiceOrURL = errors.New("webhook configuration must have either service or URL")
```

Comment:
Why is this a var? I only see one reference.

Reply:
This is moved code. fixed.